Umbraco deploy documentation (#3074)
umbraco/UmbracoDocs: Add-ons/Umbraco-Deploy/Getting-Started/Installation/New-site/index.md

---
versionFrom: 8.0.0
---

# Installing Umbraco Deploy

In this article, we will cover the steps in order for you to install Umbraco deploy on a brand new website.

We will cover how to install Umbraco deploy and set up Umbraco deploy on your website as well as show an example as to how it can be set up as a CI/CD build server using Github actions to run the deployment on a website set up with Azure Web Apps.

## Prerequisites

* Visual studio 2017 v15.9.6 or later

* Umbraco Deploy license

* SQL Server Database

## How to install Umbraco Deploy

In this guide we will show how you can install Umbraco Deploy and set up a build server using Github actions to deploy your content.

:::note

> In this example we are hosting the site on Azure Web Apps, using Github actions to set up the CI/CD build server, however you are free to choose the hosting provider and CI/CD pipeline that you prefer,
> as long as it supports executing Powershell scripts it will work with Umbraco Deploy.

Reviewer: ```suggestion
In this example we are hosting the site on Azure Web Apps, using Github actions to set up the CI/CD build server.

However you are free to choose the hosting provider and CI/CD pipeline that you prefer, as long as it supports executing Powershell scripts it will work with Umbraco Deploy.
```

---
versionFrom: 8.0.0
---

# Installing Umbraco Deploy

In this article, we will cover the steps in order for you to install Umbraco deploy on a brand new website.

> We will cover how to install Umbraco deploy and set up Umbraco deploy on your website as well as show an example as to how it can be set up as a CI/CD build server using Github actions to run the deployment on a website set up with Azure Web Apps.

Reviewer: ```suggestion
We will cover how to install Umbraco deploy and set up Umbraco deploy on your website. We will also go through an example on how it can be set up as a CI/CD build server using Github actions to run the deployment on a website set up with Azure Web Apps.
```

## Prerequisites

* Visual studio 2017 v15.9.6 or later

* Umbraco Deploy license

* SQL Server Database

## How to install Umbraco Deploy

In this guide we will show how you can install Umbraco Deploy and set up a build server using Github actions to deploy your content.

:::note
In this example we are hosting the site on Azure Web Apps, using Github actions to set up the CI/CD build server.

However you are free to choose the hosting provider and CI/CD pipeline that you prefer, as long as it supports executing Powershell scripts it will work with Umbraco Deploy.
:::

## Installation steps

1. [Set up Git repository and new Umbraco project](#Set-up-Git-repository-and-Umbraco-project)
2. [Install Umbraco Deploy via NuGet](#Installing-and-setting-up-Umbraco-Deploy)
3. [Configure CI/CD build server](#Setting-up-CI/CD-build-server-with-Github-actions)

### Set up Git repository and Umbraco project

> The first step to get Umbraco Deploy up and running is to set up a  Github repository which will act as our environment where we will set up a CI/CD pipeline that will run the build server to Azure Web Apps and new Umbraco project through [Visual Studio](https://our.umbraco.com/documentation/Getting-Started/Setup/Install/install-umbraco-with-nuget).

Reviewer: ```suggestion
The first step to get Umbraco Deploy up and running is to set up a Github repository. This will act as our environment where we will set up a CI/CD pipeline that will run the build server to Azure Web Apps and a new Umbraco project through [Visual Studio](https://our.umbraco.com/documentation/Getting-Started/Setup/Install/install-umbraco-with-nuget).
```
